app/controllers/update_username_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/importers/user_importer"
+
+class UpdateUsernameController < ApplicationController
+  before_action :require_signed_in
+  respond_to :html, :json
+  def index
+    @user = User
+  end
+
+  def update
+    @username = params['username']
+    UserImporter.update_username_for_global_id(@username)
+    redirect_to '/'
```

Comment:
Rather than redirect to the home page, I think this ought to redirect back to update username index, with a confirmation message (or a failure message, if the user couldn't be found).